controllers/warp/warp.go

```diff
@@ -67,43 +69,82 @@ func (w *Watcher) Run(ctx context.Context) error {
 	}
 
 	for _, source := range w.configuration.Sources {
-		if strings.HasPrefix(source.Path, "/dogu") {
-			w.startVersionRegistryWatch(ctx, source.Path)
-			continue
+		if source.Type == "dogu" {
+			w.startVersionRegistryWatch(ctx)
+		} else {
+			w.startGlobalConfigWatch(ctx, source.Path)
 		}
+	}
 
-		// TODO config/_global
+	return nil
+}
 
-		// TODO config/
+func (w *Watcher) startGlobalConfigWatch(ctx context.Context, sourcePath string) {
+	ctrl.LoggerFrom(ctx).Info(fmt.Sprintf("start global config watcher for source [%s]", sourcePath))
+	filter := libconfig.KeyFilter(libconfig.Key(sourcePath))
+	globalConfigWatchChannel, err := w.globalConfigRepo.Watch(ctx, filter)
+	if err != nil {
+		ctrl.LoggerFrom(ctx).Error(err, "failed to create global config watch for path %q", sourcePath)
+		return
 	}
 
-	return nil
+	go func() {
+		w.handleGlobalConfigUpdates(ctx, globalConfigWatchChannel)
+	}()
+}
 
+func (w *Watcher) handleGlobalConfigUpdates(ctx context.Context, globalConfigWatchChannel <-chan repository.GlobalConfigWatchResult) {
+	for {
+		select {
+		case <-ctx.Done():
+			ctrl.LoggerFrom(ctx).Info("context done - stop global config watch for warp generation")
+			return
+		case result, open := <-globalConfigWatchChannel:
+			if !open {
+				ctrl.LoggerFrom(ctx).Info("global config watch channel canceled - stop watch for warp generation")
+				return
+			}
+			if result.Err != nil {
+				ctrl.LoggerFrom(ctx).Error(result.Err, "global config watch channel error for warp generation")
+				continue
+			}
+			// Trigger refresh. Content of the result is not needed
+			err := w.execute(ctx)
+			if err != nil {
+				ctrl.LoggerFrom(ctx).Error(err, "failed to update entries from global config in warp menu")
+			}
+		}
+	}
 }
 
-func (w *Watcher) startVersionRegistryWatch(ctx context.Context, sourcePath string) {
-	ctrl.LoggerFrom(ctx).Info(fmt.Sprintf("start version registry watcher for source [%s]", sourcePath))
+func (w *Watcher) startVersionRegistryWatch(ctx context.Context) {
+	ctrl.LoggerFrom(ctx).Info(fmt.Sprintf("start version registry watcher for source type dogu"))
```

Comment:
🚫 **[checkstyle]** <gosimple> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>S1039: unnecessary use of fmt.Sprintf
<!-- __reviewdog__:ChA1NGE5OTU0OGVkODhjMjFmEgpjaGVja3N0eWxl -->